src/rgw/rgw_sal_motr.cc

```diff
@@ -2639,14 +2801,19 @@ int MotrAtomicWriter::complete(size_t accounted_size, const std::string& etag,
   ent.meta.owner = owner.to_str();
   ent.meta.owner_display_name = obj.get_bucket()->get_owner()->get_display_name();
   RGWBucketInfo &info = obj.get_bucket()->get_info();
-
   // Set version and current flag in case of both versioning enabled and suspended case.
   if (info.versioned())
     ent.flags = rgw_bucket_dir_entry::FLAG_VER | rgw_bucket_dir_entry::FLAG_CURRENT;
   ldpp_dout(dpp, 20) <<__func__<< ": key=" << obj.get_key().to_str()
                     << " etag: " << etag << " user_data=" << user_data << dendl;
   if (user_data)
     ent.meta.user_data = *user_data;
+
+  if (!obj.get_key().have_instance()) {
+    // generate-version-id for null version.
+    obj.gen_rand_obj_instance_name();
+    ent.key.instance = obj.get_instance();
```

Comment:
As discuss, we can directly assign "null" to key.instance in case of unversioned and suspended bucket.
Use the versionId id to create the key for kv store.

Reply:
yes, done changes as suggested.